services/minespace-web/src/components/dashboard/mine/reports/ReportsTable.js

```diff
@@ -0,0 +1,150 @@
+/* eslint-disable */
+
+import React from "react";
+import { Table, Button } from "antd";
+import moment from "moment";
+import PropTypes from "prop-types";
+import * as Strings from "@/constants/strings";
+import { formatDate } from "@/utils/helpers";
+import { EDIT_PENCIL } from "@/constants/assets";
+import CustomPropTypes from "@/customPropTypes";
+
+const propTypes = {
+  mineReports: PropTypes.arrayOf(CustomPropTypes.mineReport).isRequired,
+  openEditReportModal: PropTypes.func.isRequired,
+  handleEditReport: PropTypes.func.isRequired,
+  isLoaded: PropTypes.bool.isRequired,
+};
+
+const defaultProps = {};
+
+const columns = [
+  {
+    title: "Report Name",
+    dataIndex: "report_name",
+    key: "report_name",
+    sorter: (a, b) => a.report_name.localeCompare(b.report_name),
+    render: (text, record) => (
+      <div title="Report Name" className={record.isOverdue ? "color-error" : ""}>
+        {record.report_name}
+      </div>
+    ),
+  },
+  {
+    title: "Compliance Period",
+    dataIndex: "submission_year",
+    key: "submission_year",
+    sorter: (a, b) => (a.submission_year > b.submission_year ? -1 : 1),
+    render: (text, record) => (
+      <div title="Compliance Period" className={record.isOverdue ? "color-error" : ""}>
+        {record.submission_year}
+      </div>
+    ),
+  },
+  {
+    title: "Due",
+    dataIndex: "due_date",
+    key: "due_date",
+    sorter: (a, b) => (moment(a.due_date) > moment(b.due_date) ? -1 : 1),
+    render: (text, record) => (
+      <div title="Due" className={record.isOverdue ? "color-error" : ""}>
+        {formatDate(record.due_date) || Strings.EMPTY_FIELD}
+      </div>
+    ),
+  },
+  {
+    title: "Submitted On",
+    dataIndex: "received_date",
+    key: "received_date",
+    sorter: (a, b) => (moment(a.received_date) > moment(b.received_date) ? -1 : 1),
+    render: (text, record) => (
+      <div title="Submitted On" className={record.isOverdue ? "color-error" : ""}>
+        {formatDate(record.received_date) || Strings.EMPTY_FIELD}
+      </div>
+    ),
+  },
+  // NOTE: This is a newly requested column.
+  {
+    title: "Submitted By",
+    dataIndex: "submitted_by",
+    key: "submitted_by",
+    // sorter: (a, b) => (moment(a.received_date) > moment(b.received_date) ? -1 : 1),
+    render: (text, record) => (
+      <div title="Submitted By" className={record.isOverdue ? "color-error" : ""}></div>
+    ),
+  },
+  // NOTE: This is a newly requested column. Brian: "Update table to include this for both
```

Comment:
We should make sure to remove these notes during this sprint.

Reply:
That's what I was thinking too :)